docs/source/configurator.rst

```diff
@@ -0,0 +1,63 @@
+Configuration files and ``ebu-run``
+===================================
+
+There are 2 ways the nodes and carriage mechanisms can be set up:
+
+    1. Write a script where you instantiate these components and connect them. This technique is making use of
+    dependency injection. This means the nodes and carriage mechanisms do not need to know about the creation of
+    each other. They just get the already created component referenced as a parameter, and check if that component
+    implement the expected interface. This eliminates the possibility of creating spaghetti code where a node would know
+    how to create a carriage mechanism and deal with all the possible parameters and a carriage mechanism would know
+    in turn how to instantiate a node. This is to be avoided. And this brings us to the second way to do this.
+
+    2. Use the component configurators and use a simple configuration file to create your system and give the
+    configuration file to the :py:mod:`ebu_tt_live.scripts.ebu_run` script, which takes care of creating all the
+    required components for you. This eliminates the need for a programmer to keep repeating the configuration logic
+    in their scripts where they create the components.
+
+The idea is the following: The components(nodes, carriage mechanisms, converters, resegmenters...etc.) are agnostic
+to one another as long as the carriage mechanism is capable of transporting the payload the node expects. But their
+wiring and their factories are actually the configurator classes. The configurator class defines the type and
+name of the configuration options relevant to the component and wires it with its immediate dependencies. This
+way the factories that create the system are separate from the system's business logic and this separation is
+meant to be kept. There is a simple naming correspondence. For instance the business logic of the processing
```

Comment:
because one of them is a package, and the other one is a module. This is what they are called in python.